src/core/server/status/status_service.ts

```diff
@@ -18,39 +18,62 @@
  */
 
 import { Observable, combineLatest } from 'rxjs';
-import { map, distinctUntilChanged, shareReplay, take } from 'rxjs/operators';
+import { map, distinctUntilChanged, shareReplay, take, debounceTime } from 'rxjs/operators';
 import { isDeepStrictEqual } from 'util';
 
 import { CoreService } from '../../types';
 import { CoreContext } from '../core_context';
 import { Logger } from '../logging';
 import { InternalElasticsearchServiceSetup } from '../elasticsearch';
 import { InternalSavedObjectsServiceSetup } from '../saved_objects';
+import { PluginName } from '../plugins';
 
 import { config, StatusConfigType } from './status_config';
 import { ServiceStatus, CoreStatus, InternalStatusServiceSetup } from './types';
 import { getSummaryStatus } from './get_summary_status';
+import { PluginsStatusService } from './plugins_status';
 
 interface SetupDeps {
   elasticsearch: Pick<InternalElasticsearchServiceSetup, 'status$'>;
+  pluginDependencies: ReadonlyMap<PluginName, PluginName[]>;
   savedObjects: Pick<InternalSavedObjectsServiceSetup, 'status$'>;
 }
 
+const waitForMicrotasks = (numberOfTicks: number = 1): Promise<unknown> => {
```

Comment:
can be removed?

Reply:
heh, TIL that TS doesn't complain about an unused function if the only use is a recursive call to itself